azurelinuxagent/common/utils/restutil.py

```diff
@@ -145,55 +213,123 @@ def http_request(method, url, data, headers=None, max_retry=3,
                                  headers=headers,
                                  proxy_host=proxy_host,
                                  proxy_port=proxy_port)
-            logger.verbose("HTTP response status: [{0}]", resp.status)
+            logger.verbose("[HTTP Response] Status Code {0}", resp.status)
+
+            if request_failed(resp):
+                if resp.status in retry_codes:
+                    msg = '[HTTP Retry] HTTP {0} Status Code {1}'.format(
+                        method, resp.status)
+                    if resp.status in THROTTLE_CODES:
+                        delay = LONG_DELAY_IN_SECONDS
+                        logger.info("[HTTP Delay] Delay {0} seconds for " \
+                                    "Status Code {1}".format(
+                                        delay, resp.status))
+                    continue
+
+            if resp.status == httpclient.BAD_REQUEST:
+                raise BadRequestError()
+
             return resp
+
         except httpclient.HTTPException as e:
-            retry_msg = 'HTTP exception: {0} {1}'.format(log_msg, e)
-            retry_interval = 5
+            msg = '[HTTP Failed] HTTP {0} Exception {1}'.format(method, e)
+            if len([x for x in RETRY_EXCEPTIONS if isinstance(e, x)]) > 0:
```

Comment:
>len([x for x in RETRY_EXCEPTIONS if isinstance(e, x)]) > 0: [](start = 15, length = 59)

I like your request_failed() and request_succeeded() methods, so why not add one for this too.  A name is better than a list comprehension.